crates/anstyle-roff/src/lib.rs

```diff
@@ -0,0 +1,152 @@
+//! Convert from ansi stylings to ROFF Control Lines
+//! Currently uses [roff](https://docs.rs/roff/0.2.1/roff/) as the engine for generating
+//! roff output.
+
+mod styled_str;
+use anstyle::{Color, RgbColor};
+use roff::{bold, italic, Roff};
+use styled_str::StyledStr;
+
+/// Generate A RoffStyle from Style
+///
+/// ```rust
+/// use anstyle::{Color, RgbColor};
+///
+/// let text = "\u{1b}[44;31mtest\u{1b}[0m";
+///
+/// let roff_doc = anstyle_roff::to_roff(text);
+/// let expected = r#".gcolor red
+/// .fcolor blue
+/// test
+/// "#;
+///
+/// assert_eq!(roff_doc.to_roff(), expected);
+/// ```
+pub fn to_roff(styled_text: &str) -> Roff {
+    let mut roff_docs = vec![];
+    for styled in styled_str::styled_stream(styled_text) {
+        roff_docs.push(as_roff(&styled))
+    }
+
+    let mut doc = Roff::new();
+    doc.extend(roff_docs);
+    doc
+}
+
+fn as_roff(styled: &StyledStr) -> Roff {
+    let style = styled.style;
+    let mut doc = Roff::new();
+    doc.extend([
+        set_color((&style.get_fg_color(), &style.get_bg_color())),
+        set_effects(styled),
+    ]);
+    doc
+}
+
+fn set_effects(styled: &StyledStr) -> Roff {
+    // Roff (the crate) only supports these inline commands
+    //  - Bold
+    //  - Italic
+    //  - Roman (plain text)
+    // If we want more support, or even support combined formats, we will need
+    // to push improvements to roff upstream or implement a more thorough roff crate
+    // perhaps by spinning off some of this code
+    let effects = styled.style.get_effects();
+    let mut doc = Roff::new();
+
+    if effects.contains(anstyle::Effects::BOLD) {
+        doc.text(vec![bold(styled.text)]);
+        return doc;
+    }
+
+    if effects.contains(anstyle::Effects::ITALIC) {
+        doc.text(vec![italic(styled.text)]);
+        return doc;
+    }
+
+    if effects.is_plain() {
+        doc.text(vec![roff::roman(styled.text)]);
+        return doc;
+    }
+
+    doc
```

Comment:
It looks like this is dropping unsupported text, which seems bad.

Reply:
Hmm. yeah I don't know why i didn't go with a plain else here.